qctrlopencontrols/dynamic_decoupling_sequences/predefined.py

```diff
@@ -416,14 +411,15 @@ def new_cpmg_sequence(
     .. [#] `S. Meiboom and D. Gill, Review of Scientific Instruments 29:8, 688 (1958).
         <https://link.aps.org/doi/10.1063/1.1716296>`_
     """
-    duration = _check_duration(duration)
-    number_of_offsets = number_of_offsets or 1
-    number_of_offsets = int(number_of_offsets)
```

Comment:
This line `number_of_offsets = int(number_of_offsets)` used to prevent a user from mistakenly passing something like `number_of_offsets = 3.000001` which I think would cause the pulses to exceed the duration of the sequence.

I'm not entirely sure if this line is necessary, as the docstring says that the user should pass an integer for this parameter. But in any case, just wanted to know your thoughts about this

Reply:
That's a good question. In that case, would that mean `number_of_offsets` is calculated from other functions, which normally is a `float` number in python? 

The reason I removed is the doc says it should be `int`, as you mentioned. But it's python, we can't really stop people from passing a float number. Adding this line back seems to be safe. Maybe we can do that for now and ask Harry on Monday see whether there is a better way?

Reply:
Sounds good

Reply:
I made the change in the private function as it seems we always calculate offset from there. In that way, we can hide this type issue from the main function.

Reply:
I'm okay with hiding it in the private function, but I think at the moment a user mistakenly passing `number_of_offsets = 3.000001` will still get offsets that go beyond the duration of the sequence, because `np.arange(3.000001)` is the same as `np.arange(4)`. Maybe we can write `spacing * np.arange(int(number_of_offsets))` in the private function?

Reply:
ah, sorry, clearly I missed that part. Will fix. Thanks for pointing it out!